### PR TITLE
New MvxPageViewController

### PR DESCRIPTION
**Pull request by chrilith**
Hi,

I've just commited a new implementation of the MvxPageViewController which works more like any other controllers in MvvmCross.

The current implementation, based on view models, is really interesting, but one thing that is hardly missing is binding (AFAICT) and a way to refresh the controller. Also, as stated in #1403 we are forced to use an internal cache, when iOS already cache things on its own.

Here, I added a simple data source that implements most of the minimal requirements to  properly handle PageViews.

I will add some events later if this PR is acceptable to the dev team. Don't want to waste too much time right now because i have an app to develop :)

_Added_:
- `MvxPageViewSource`: the important part here is `GetViewControllerAtIndex()` which will provide the expected view conroller if any and `GetPageIndexForController()`.
- `IMvxpageViewModel` and `IMvxPageViewController`. Both define à `PageIndex` that will be used by the default data source (`MvxPageViewSource.GetPageIndexForController()`)

Basically everything works with ViewControllers with iOS page views so I think it is important to keep this logic.

_Moved_:
- Old code has been moved to a Legacy namespace to keep compatibility. I can add a new source that mimic most of the old behavior if needed.

Let me know if you have any questions.
Chris

### CODE REVIEW

**Comment by Cheesebaron** [comment]
Can you please merge master into this branch. Will have a look when you do that 👍

**Comment by chrilith** [comment]
OK, will try to do it this week!

File: MvvmCross/Binding/iOS/Views/MvxBasePageViewSource.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        protected UIPageViewController PageView => this._pageView;
+
+        public virtual void ReloadData()
+        {
+            try
+            {
+                var controller = GetInitialViewController();
+                var list = new UIViewController[] { controller };
+                // FIXME: we are forced here to have animated:false because of a bug in iOS up to 9.x
```
Comment (martijn00) [inline MvvmCross/Binding/iOS/Views/MvxBasePageViewSource.cs]: Maybe do a check for the iOS version and put that in a bool variable. Then use that to animate on iOS 9 and higher?

**Comment by martijn00** [comment]
I wouldn't leave the legacy code in. We can remove that if this PR gets accepted and tested, and release this as a breaking change for MvvmCross 5.0 (https://github.com/MvvmCross/MvvmCross/issues/1415)

File: MvvmCross/Binding/iOS/Views/MvxBasePageViewSource.cs (hunk 1 above)
Comment (chrilith) [inline MvvmCross/Binding/iOS/Views/MvxBasePageViewSource.cs]: OK, but I have to check if this bug have been fixed in iOS 10 first. If fixed, I'll add the check.

**Comment by chrilith** [comment]
@martijn00 OK, will remove the legacy code then.

**Comment by chrilith** [comment]
Did a rebase. Will work then on Legacy removal and check the animated:false problem.

**Comment by chrilith** [comment]
Should be up to date and clean now.

**Comment by martijn00** [comment]
It seems that your merge of upstream didn't went in correct. I can see the changes in the files in this PR.

**Comment by martijn00** [comment]
Is there a sample repo for this available?

**Comment by chrilith** [comment]
Will do a quick sample.

**Comment by martijn00** [comment]
How is the sample coming along? Can you also merge the upstream into your branch?

**Comment by chrilith** [comment]
Will do all this on monday!

**Comment by chrilith** [comment]
Hum, why are there conflicts?

Here is the PoC: https://www.dropbox.com/s/3qqayvku6028h6l/MVVMCross_PageViewPoC.zip?dl=0

**Comment by Cheesebaron** [comment]
Someone changed some csproj files and MvxPageViewController outside of this PR and merged into develop. Hence there are conficts. You need to merge develop into your branch or rebase.

**Comment by chrilith** [comment]
OK, did a merge but the solution doesn't compile with Visual Studio. I have no portable target for tvOS and so I cannot update the new tvOS stuff.

Any clue?

**Comment by Cheesebaron** [comment]
I'll try make a new branch and cherry pick your changes, because this looks very messed up. Lets get this into MvvmCross 😃

**Comment by chrilith** [comment]
OK thanks :( 
Do you know how to get tvOS target in Visual Studio?

**Comment by Cheesebaron** [comment]
You shouldn't have to do anything. It won't appear in a PCL as targets though.

If you need to create an actual tvOS project, I think you need to do it on a Mac.

**Comment by chrilith** [comment]
Back, will try to reset and add again.

**Comment by chrilith** [comment]
OK, my branch is totally messed up. Let's close this PR and create a new one as proposed by Cheesebaron.
